### PR TITLE
Chore: Reduce settings Go code complexity

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
As part of enabling gocyclo linter, make Go code to load settings less complex (by breaking huge functions into more manageable ones).

I've done basic manual testing and it works.

Probably easiest diffed locally with a GUI tool.

### CODE REVIEW

**Review by kylebrandt — APPROVED** [review]

**Review by ryantxu — APPROVED** [review]
I am all for it... and trust you that this is better ;)

File: pkg/setting/setting.go [hunk 1]
```diff
@@ -661,73 +661,264 @@ func (cfg *Cfg) Load(args *CommandLineArgs) error {
 		return err
 	}
 	cfg.ProvisioningPath = makeAbsolute(provisioning, HomePath)
-	server := iniFile.Section("server")
-	AppUrl, AppSubUrl, err = parseAppUrlAndSubUrl(server)
+	if err := readServerSettings(iniFile, cfg); err != nil {
+		return err
```
Comment (marefr) [inline pkg/setting/setting.go:665]: Nit: Maybe return some context when returning errors from reading certain sections/settings by wrapping error in "failed to read <section/setting> settings" or similar?

**Review by marefr — APPROVED** [review]
LGTM. Added a nit comment

File: pkg/setting/setting.go (hunk 1 above)
Comment (aknuds1) [reply pkg/setting/setting.go:665]: @marefr Isn't that out of scope for the PR, since all it does is refactor?

Comment (marefr) [reply pkg/setting/setting.go:665]: Sure

**Pull request merged**